src/v/cloud_storage/partition_manifest.cc

```diff
@@ -814,6 +848,10 @@ partition_manifest partition_manifest::truncate() {
         _start_offset = model::offset{};
         // NOTE: _last_offset should not be reset
     }
+    // Update size of the archived part of the log.
+    // It doesn't include segments which are remaining in the
+    // manifest.
+    _archive_size_bytes += removed.cloud_log_size();
```

Comment:
I think this addition breaks retention and is the cause of the `Can't truncate up to` test failures. `partition_manifest::cloud_log_size` returns `_cloud_log_size_bytes + _archive_size_bytes`. This means that whenever `partition_manifest::truncate` is called to handle a `cleanup_metadata_cmd` the reported size remains the same and retention kicks in and fails because there's no segments to remove.

Reply:
Fixed this by introducing new code path for spillover.

Reply:
I'm not seeing the new path. The handling for both `cleanup_metadata_cmd` and `truncate_cmd` still shares the call to `partition_manifest::truncate` which updates `_archive_size_bytes`.